src/Microsoft.AspNet.Mvc.Core/ModelBinding/HeaderModelBinder.cs

```diff
@@ -36,7 +36,7 @@ protected override Task<ModelBindingResult> BindModelCoreAsync([NotNull] ModelBi
             object model = null;
             if (bindingContext.ModelType == typeof(string))
             {
-                var value = request.Headers.Get(headerName);
+                string value = request.Headers[headerName];
                 if (value != null)
```

Comment:
The implicit string converter can return null. Same with the implicit array converter.